examples/dat3m/campaign_dat3m.py

```diff
@@ -0,0 +1,72 @@
+#!/usr/bin/env python3
+# Copyright (C) 2024 Vrije Universiteit Brussel. All rights reserved.
+# SPDX-License-Identifier: MIT
+
+from bench import Dat3mBench
+from dat3m import standalone_dat3m_builder
+from pythainer.runners import ConcreteDockerRunner
+
+from benchkit.campaign import CampaignCartesianProduct, CampaignSuite
+from benchkit.communication.docker import DockerCommLayer
+from benchkit.platforms import Platform
+
+
+def get_local_docker_platform(docker_runner: ConcreteDockerRunner) -> Platform:
+    docker_comm = DockerCommLayer(docker_runner=docker_runner)
+    platform = Platform(comm_layer=docker_comm)
+    return platform
+
+
+def get_dat3m_docker_platform() -> Platform:
+    dat3m_builder = standalone_dat3m_builder()
+    # TODO building images does not support yet remote, the plan would be to give an optional shell_out callback
+    # to that build() function and provide with the platform.shell from the selected host.
+    dat3m_builder.build()
+    dar3m_runner = dat3m_builder.get_runner()
+    platform = get_local_docker_platform(docker_runner=dar3m_runner)
+    return platform
+
+
+def main():
+    nb_runs = 3
+
+    dat3m_platform = get_dat3m_docker_platform()
+
+    dat3m_bench = Dat3mBench(
+        src_dir="",  # TODO src_dir to mount in docker to access user client codes?
+        platform=dat3m_platform,
+    )
+
+    campaign = CampaignCartesianProduct(
+        name="dat3mlocks",
+        benchmark=dat3m_bench,
+        nb_runs=nb_runs,
+        variables={
+            "memory_model": ["aarch64"],
+            "target_arch": ["arm8"],
+            "lock_name": list(dat3m_bench.dat3m_locks),
+            "bound": [None, 1, 3],
```

Comment:
I assume `None` avoids explicitly adding the `--bound=...` option to dat3m cmd. In such a case, the default value is 1, and thus having `None` and `1` is redundant.

Reply:
I see. I will keep it supported as it is but I can remove it from the campaign.